### PR TITLE
Improve use keyspace queries detection

### PR DESCRIPTION
**Pull request by havaker**
This PR removes error prone CQL parsing in `session::query` method in a way described in #168

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [x] I added relevant tests for new features and bug fixes.
- [x] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [x] I added appropriate `Fixes:` annotations to PR description.

Fixes: #168

### CODE REVIEW

**Review by piodul — CHANGES_REQUESTED** [review]
Left some minor comments, but looks good otherwise.

File: docs/source/queries/usekeyspace.md [hunk 1]
```diff
@@ -49,6 +49,17 @@ session
 The first argument is the keyspace name.  
 The second argument states whether this name is case sensitive.
 
+It is also possible to send raw use keyspace query using `Session::query` instead of `Session::use_keyspace` such as:
```
[the remark this answers is not in the document]
Comment (havaker) [reply docs/source/queries/usekeyspace.md:52]: Done

File: scylla/src/transport/session.rs [hunk 2]
```diff
@@ -431,39 +431,36 @@ impl Session {
...
+            .await?;
+
+        if let Some(set_keyspace) = response.as_set_keyspace() {
+            debug!(
+                "Detected USE KEYSPACE query, setting session's keyspace to {}",
+                set_keyspace.keyspace_name
+            );
+            self.use_keyspace(set_keyspace.keyspace_name.clone(), true)
+                .await?;
+        }
```
[the remark this answers is not in the document]
Comment (havaker) [reply scylla/src/transport/session.rs:470]: Done

**Comment by havaker** [comment]
v2: applied suggestions from @piodul's review

**Review by piodul — APPROVED** [review]

**Pull request merged**